packages/shared/src/hooks/usePersonalizedDigest.ts

```diff
@@ -51,16 +55,21 @@ export const usePersonalizedDigest = (): UsePersonalizedDigest => {
   );
 
   const { mutateAsync: subscribePersonalizedDigest } = useMutation(
-    async () => {
+    async (params: { hour?: number; type?: UserPersonalizedDigestType }) => {
+      const { hour, type } = params || {
+        hour: 8,
+        type: UserPersonalizedDigestType.Digest,
+      };
       const result = await request<
         {
           subscribePersonalizedDigest: UserPersonalizedDigest;
         },
         Partial<UserPersonalizedDigestSubscribe>
       >(graphqlUrl, SUBSCRIBE_PERSONALIZED_DIGEST_MUTATION, {
         day: 3,
-        hour: 8,
+        hour,
```

Comment:
We need to make sure we fallback to `8` since that is the current default for digest.

Reply:
Yeah it's done in the abstraction above.

```js
const { hour, type } = params || {
        hour: 8,
        type: UserPersonalizedDigestType.Digest,
      };
```

Reply:
If someone passes `{ type: 'xxx' }` `hour` would still be undefined

Reply:
Ah you are right indeed.
There's no direct way to do it right?

Eg we need to do it down the line based on ?? fallback per both?

Reply:
I would do it like:
```js
const { hour = 8, type = UserPersonalizedDigestType.Digest } = params || {}
```

but `??` also works.